language-server/src/server.js

```diff
@@ -247,13 +251,26 @@ const validateSchema = async (textDocument) => {
       } else {
         diagnostics.push(buildDiagnostic(schemaInstance, "No dialect"));
       }
-
       continue;
     }
 
     try {
       const annotatedInstance = await annotate(dialectUri, schemaInstance);
 
+      const references = [];
+      for (const { keywordInstance } of getSemanticTokens(schemaResources)) {
+        const found = findRef(keywordInstance, dialectUri);
+        if (typeof found !== "undefined") {
+          references.push(found);
+        }
+      }
+      for (const ref of references) {
+        const output = await validateReference(documents, textDocument, ref);
+        if (!output.valid) {
+          const instance = JsoncInstance.fromTextDocument(textDocument).get("#" + ref);
```

Comment:
Do you really need to construct a whole new instance here? Can you just do `schemaResource.get`?